### PR TITLE
Implements magma-orc8r-lite charms and bundle

### PR DESCRIPTION
**Pull request by gruyaume**

### CODE REVIEW

File: orchestrator-bundle/orc8r-controller-operator/tests/integration/test_integration.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+APPLICATION_NAME = "orc8r-controller"
+CHARM_NAME = "magma-orc8r-controller-lite"
+
+
+class TestOrc8rcontroller:
+    @pytest.fixture(scope="module")
+    async def setup(self, ops_test):
+        await ops_test.model.deploy("postgresql-k8s", application_name="postgresql-k8s")
+        await ops_test.model.wait_for_idle(apps=["postgresql-k8s"], status="active", timeout=1000)
```
Comment (sanchezfdezjavier) [inline orchestrator-bundle/orc8r-controller-operator/tests/integration/test_integration.py:23]: As suggested by Simon in the `prometheus-edge-hub` PR. We might consider using `asyncio.gather`.
```suggestion
	await asyncio.gather(
		ops_test.model.deploy("postgresql-k8s", application_name="postgresql-k8s")
		ops_test.model.wait_for_idle(apps=["postgresql-k8s"], status="active", timeout=1000)
	)
```

File: orchestrator-bundle/orc8r-controller-operator/tests/integration/test_integration.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            charm,
+            resources=resources,
+            application_name=APPLICATION_NAME,
+            config={"domain": "example.com"},
+            trust=True,
+        )
+        await ops_test.model.add_relation(
+            relation1=APPLICATION_NAME, relation2="postgresql-k8s:db"
+        )
+        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1000)
```
Comment (sanchezfdezjavier) [inline orchestrator-bundle/orc8r-controller-operator/tests/integration/test_integration.py:41]: Same here.
```suggestion
	await asyncio.gather(
	          ops_test.model.deploy(
	              charm,
	              resources=resources,
	              application_name=APPLICATION_NAME,
	              config={"domain": "example.com"},
	              trust=True,
	          )
	          ops_test.model.add_relation(
	              relation1=APPLICATION_NAME, relation2="postgresql-k8s:db"
	          )
	          ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1000)
	)
```

File: orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+CHARM_NAME = "magma-orc8r-nginx"
+CONTROLLER_APPLICATION_NAME = "orc8r-controller"
+CONTROLLER_CHARM_NAME = "magma-orc8r-controller"
+
+
+class TestOrc8rNginx:
+    @pytest.fixture(scope="module")
+    async def setup(self, ops_test):
+        await self._deploy_postgresql(ops_test)
+        await self._deploy_orc8r_controller(ops_test)
```
Comment (sanchezfdezjavier) [inline orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py]: Same here.
```suggestion
    async def setup(self, ops_test):
    	await asyncio.gather(
		self._deploy_postgresql(ops_test)
		self._deploy_orc8r_controller(ops_test)
	)
```

File: orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        resources = {
+            f"{CHARM_NAME}-image": METADATA["resources"][f"{CHARM_NAME}-image"]["upstream-source"],
+        }
+        await ops_test.model.deploy(
+            charm, resources=resources, application_name=APPLICATION_NAME, trust=True
+        )
+        await ops_test.model.add_relation(
+            relation1=APPLICATION_NAME, relation2="orc8r-controller:controller"
+        )
+        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1000)
```
Comment (sanchezfdezjavier) [inline orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py]: Same here.
```suggestion
	await asyncio.gather(
	       	  ops_test.model.deploy(
	              charm, resources=resources, application_name=APPLICATION_NAME, trust=True
	          )
	          ops_test.model.add_relation(
	              relation1=APPLICATION_NAME, relation2="orc8r-controller:controller"
	          )
	          ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1000)
	)
```

File: orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+    @staticmethod
+    async def _deploy_postgresql(ops_test):
+        await ops_test.model.deploy("postgresql-k8s", application_name="postgresql-k8s")
+        await ops_test.model.wait_for_idle(apps=["postgresql-k8s"], status="active", timeout=1000)
```
Comment (sanchezfdezjavier) [inline orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py:50]: Same here. I think you've already got my point 😆
```suggestion
	await asyncio.gather(
		ops_test.model.deploy("postgresql-k8s", application_name="postgresql-k8s")
		ops_test.model.wait_for_idle(apps=["postgresql-k8s"], status="active", timeout=1000)
	)
```

File: orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            application_name=CONTROLLER_APPLICATION_NAME,
+            config={"domain": "example.com"},
+            trust=True,
+        )
+        await ops_test.model.add_relation(
+            relation1=CONTROLLER_APPLICATION_NAME, relation2="postgresql-k8s:db"
+        )
+        await ops_test.model.wait_for_idle(
+            apps=[CONTROLLER_APPLICATION_NAME], status="active", timeout=1000
+        )
```
Comment (sanchezfdezjavier) [inline orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py:72]: And finally, the last one.
```suggestion
	await asyncio.gather(
		ops_test.model.deploy(
	              charm,
	              resources=resources,
	              application_name=CONTROLLER_APPLICATION_NAME,
	              config={"domain": "example.com"},
	              trust=True,
	          )
		ops_test.model.add_relation(
	              relation1=CONTROLLER_APPLICATION_NAME, relation2="postgresql-k8s:db"
	          )
		ops_test.model.wait_for_idle(
	              apps=[CONTROLLER_APPLICATION_NAME], status="active", timeout=1000
	          )
	)
```

File: orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import yaml
+from pytest_operator.plugin import OpsTest  # type: ignore[import]  # noqa: F401
+
+logger = logging.getLogger(__name__)
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+CONTROLLER_METADATA = yaml.safe_load(
+    Path("../orc8r-controller-operator/metadata.yaml").read_text()
+)
+APPLICATION_NAME = "orc8r-nginx"
+CHARM_NAME = "magma-orc8r-nginx"
```
Comment (sanchezfdezjavier) [inline orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py]: Shouldn't be the `CHARM_NAME` `magma-orc8r-nginx-lite` as it is in `metadata.yaml`. Maybe I'm missing something.

**Review by sanchezfdezjavier — COMMENTED** [review]
Should we use this to test `charmed-magma` easily with images built locally? Really cool idea to make this version 🎉

File: orchestrator-bundle/orc8r-controller-operator/tests/integration/test_integration.py (hunk 1 above)
Comment (gruyaume) [reply orchestrator-bundle/orc8r-controller-operator/tests/integration/test_integration.py:23]: The difference was that in prometheus-edge-hub we were awaiting two independant deployments at the same time. In our case it's a bit diferent. We need to deploy before waiting for idle. There's an order to follow. The same goes for all the comments in the PR.

File: orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py (hunk 3 above)
Comment (gruyaume) [reply orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py]: Here we can bundle them with gather.

Comment (gruyaume) [reply orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py]: Fixed

File: orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py (hunk 7 above)
Comment (gruyaume) [reply orchestrator-bundle/orc8r-nginx-operator/tests/integration/test_integration.py]: Yah I changed the variable names as it was confusing. The image name is `magma-orc8r-nginx-image`.

File: orchestrator-bundle/orc8r-controller-operator/tests/integration/test_integration.py (hunk 1 above)
Comment (sanchezfdezjavier) [reply orchestrator-bundle/orc8r-controller-operator/tests/integration/test_integration.py:23]: Yep, my bad! Thanks for clarifying :)

**Review by sanchezfdezjavier — APPROVED** [review]

**Pull request merged**